### DISCUSSION TITLE
Changing config file to regkey

### DISCUSSION THREAD

**Discussion by Romanitho** [Romanitho/Winget-AutoUpdate#76] [Polls]
Wondered if it was not better to create regkey like standard app to store config and app version for instance. This way, it is easier to detect with SCCM or Intune and to manage WAU when self update is disabled.

**Comment by Romanitho** [comment]
something like:  
HKEY_LOCAL_MACHINE\SOFTWARE\WAU
or  
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\WAU

  
| Name | Type | Data |
| --- | --- | --- |
| DisplayName | REG_SZ | Winget-AutoUpdate (WAU) |
| DisplayVersion | REG_SZ | 1.10.0 |
| Publisher | REG_SZ | Romanitho |
| URLInfoAbout | REG_SZ | https://github.com/Romanitho/Winget-AutoUpdate |
| NotificationLevel | REG_SZ | Full |
| WAUautoupdate | REG_DWORD | 1 |
| WAUprerelease | REG_DWORD | 0 |
| UseWAUWhiteList | REG_DWORD | 0 |

**Reply by KnifMelti** [reply]
Maybe (in `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\WAU `too):
- Contact
- InstallDate
EDIT: Deleted - Uninstallstring (REG_EXPAND_SZ)

**Reply by Romanitho** [reply]
Yes, I am thinking about remaking the update and uninstall part. This way, we can run separate scripts to update or clean and have proper registry entries.

**Comment by KnifMelti** [comment]
But it doesn't matter where you put it, under `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\WAU` it doesn't populate the Programs and Features anyway...

**Reply by Romanitho** [reply]
It does actually, and uninstall feature works 😅  
working on v1.11.0

**Reply by KnifMelti** [reply]
then.. ..DisplayIcon (REG_SZ) too :)

**Reply by KnifMelti** [reply]
I must have tested hastily and poorly...

**Reply by KnifMelti** [reply]
And VersionMajor/VersionMinor (REG_DWORD) so in SCCM we can have detection on **Version >= 1.11**

**Reply by KnifMelti** [reply]
...but, the Version in SCCM seems to check **Version** (REG_DWORD) from a MSI installation https://docs.microsoft.com/en-us/windows/win32/msi/uninstall-registry-key

**Reply by KnifMelti** [reply]
...and that is the version in HEX from Major, Minor, Build (1.11.0) = 1B0

**Reply by Romanitho** [reply]
How I use my detection method now:  
![image](https://user-images.githubusercontent.com/96626929/169975532-844cf408-01da-470d-8515-d439a07db2b1.png)

**Comment by Romanitho** [comment]
v1.11.0 is out as pre-release

**Reply by KnifMelti** [reply]
So, this is your official icon `"C:\Windows\System32\shell32.dll,-16739"` - I think it's used in Windows as an icon for Windows Update...
...should I change in our SCCM to this?

**Reply by Romanitho** [reply]
it's just a random icon a picked up to represent update things... haha. don't have any icon (yet).
it was to avoid blank icon in program and feature.
